### PR TITLE
ref(hybrid-cloud): make notification_settings.actor_id nullable

### PR DESCRIPTION
**Pull request by mikejihbe**

### CODE REVIEW

File: src/sentry/services/hybrid_cloud/notifications/service.py [hunk 1]
```diff
@@ -106,6 +110,20 @@ def remove_notification_settings_for_user(
     ) -> None:
         pass
 
+    @rpc_method
+    @abstractmethod
+    def filter(
```
Comment (markstory) [inline src/sentry/services/hybrid_cloud/notifications/service.py:115]: Where is the implementation for this method? I don't see it in master.

File: src/sentry/notifications/manager.py [hunk 2]
```diff
@@ -404,27 +400,23 @@ def update_settings_bulk(
...
         user_ids: Set[int] = set()
-        (team_ids if actor.actor_type == ActorType.TEAM else user_ids).add(actor.id)
 
-        target_id = actor.actor_id
-        assert target_id, "Cannot update settings for None actor_id"
+        if user is not None:
+            user_ids.add(user.id)
+            id_key = "user_id"
+            id = user.id
+        if team is not None:
```
Comment (markstory) [inline src/sentry/notifications/manager.py:415]: Should there be an error if both team and user are None?

File: src/sentry/notifications/manager.py [hunk 3]
```diff
@@ -437,21 +431,19 @@ def update_settings_bulk(
...
-                    **{id_key: actor.id},
+                    **{id_key: id},
                 )
         analytics.record(
             "notifications.settings_updated",
-            target_type="user" if actor.actor_type == ActorType.USER else "team",
-            actor_id=target_id,
-            id=actor.id,
+            target_type="user" if user is not None else "team",
+            actor_id=None,
```
[the remark this answers is not in the document]
Comment (scefali) [reply src/sentry/notifications/manager.py:445]: @mikejihbe nope don't think so but this analytics event is now useless

[the remark this answers is not in the document]
Comment (mikejihbe) [reply src/sentry/notifications/manager.py:445]: Nah, you still have the type and the user or team ID in there, you'd just query for it differently

**Review by markstory — APPROVED** [review]

File: src/sentry/notifications/manager.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (scefali) [reply src/sentry/notifications/manager.py:445]: @mikejihbe ok it won't work for Amplitude but I guess it will work ok for Redash

**Pull request merged**